src/defargs.l

```diff
@@ -501,9 +517,54 @@ CPPC  "/\/"
 <CopyArgRound,CopyArgRound2,CopyArgSharp,CopyArgCurly>.  {
 					  *yyextra->copyArgValue += *yytext;
 					}
-<ReadTypeConstraint>[,)]               {
-                                          unput(*yytext);
-                                          BEGIN(yyextra->lastExtendsContext);
+<ReadAnnotation>"("                    {
+                                          ++yyextra->argRoundCount;
+                                       }
+<ReadAnnotation>")"                    {
+                                          if (--yyextra->argRoundCount == 0)
+                                          {
+                                            BEGIN(yyextra->readAnnotationContext);
+                                          }
+                                       }
+<ReadAnnotation>{B}+                   {
+                                          if (yyextra->argRoundCount == 0)
+                                          {
+                                            BEGIN(yyextra->readAnnotationContext);
+                                          }
+                                       }
+<ReadAnnotation>"\""                   {
+                                          BEGIN(ReadStringLiteral);
+                                       }
```

Comment:
Does for annotations also the single quote possibility exist?

Reply:
No. The string literals in annotations are just usual Java string literals. Java *only* uses double quotes as string boundaries, much like C and friends do (and unlike Python does).

Reply:
Oh sorry, I forgot there can also be character literals, which indeed use a single quote. However, there can only be one single character in between (or an escape sequence), so it should be sufficient to handle the rules ``<ReadAnnotation>"'('"`` and ``<ReadAnnotation>"')'"``, ``<ReadAnnotation>"'\"'"`` and probably also ``<ReadAnnotation>"'\\\"'"``. It should be possible to combine the first three into ``<ReadAnnotation>"'"[()\"]"'"``. I will add them soon.

Reply:
Not sure but don't forget about the `'\''` i.e. "the escaped single quote character literal".

Reply:
I thought about that, but it should be covered by the general ``<ReadAnnotation>.`` rule. Since we do not enter a new state on ``'`` (unlike we do for ``"``), no extra care should be required to avoid premature return from such a state. The ``"`` must be considered to avoid entering ``<ReadStringLiteral>`` falsely, which cannot happen for ``'``.

Reply:
But couldn't in that case `'"`'` be problematic?

Reply:
That's covered by ``<ReadAnnotation>"'\"'"``. The overall rule I added just now is

```<ReadAnnotation>"'"[()\"]"'"|"'\\\"'"```. It should catch the Java literals ``'('``, ``')'``, ``'"'``, ``'\"'`` and I confirmed this with some simple tests.